test/data/test_builtin_datasets.py

```diff
@@ -47,3 +49,25 @@ def test_penntreebank(self):
         if os.environ.get("TRAVIS") == "true":
             datafile = os.path.join(self.project_root, ".data", "penn-treebank")
             conditional_remove(datafile)
+
+    def test_text_classification(self):
+        # smoke test to ensure ag_news dataset works properly
+        datadir = os.path.join(self.project_root, ".data")
+        txt_cls = TextClassificationDataset("ag_news", root=datadir, ngrams=2)
+        train_iter, test_iter, valid_iter = txt_cls.iters(batch_size=128,
+                                                          device="cpu")
+        ag_news_cls = AG_NEWS(root=datadir, ngrams=3)
+        train_iter, test_iter, valid_iter = ag_news_cls.iters(batch_size=128,
+                                                              device="cpu")
+
+        # Delete the dataset after we're done to save disk space on CI
+        if os.environ.get("TRAVIS") == "true":
+            datafile = os.path.join(self.project_root,
```

Comment:
Instead of this you should be able to create a temporary directory via https://docs.python.org/3/library/tempfile.html , which has all of this built in and is therefore safer.